src/Microsoft.CodeQuality.Analyzers.Exp/Core/Maintainability/ReviewSqlQueriesForSecurityVulnerabilities.cs

```diff
@@ -0,0 +1,249 @@
+
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftMaintainabilityAnalyzersResources.ReviewSQLQueriesForSecurityVulnerabilitiesDescription), MicrosoftMaintainabilityAnalyzersResources.ResourceManager, typeof(MicrosoftMaintainabilityAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageNoNonLiterals,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: "https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2100-review-sql-queries-for-security-vulnerabilities",
+                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            INamedTypeSymbol iDbCommandType = null;
+            INamedTypeSymbol iDataAdapterType = null;
+            IPropertySymbol commandTextProperty = null;
+
+            // Just getting the well known types once, no actual actions to take
+#pragma warning disable RS1012 // Start action has no registered actions.
+            context.RegisterCompilationStartAction(compilationContext =>
+#pragma warning restore RS1012 // Start action has no registered actions.
+            {
+                iDbCommandType = WellKnownTypes.IDbCommand(compilationContext.Compilation);
+                iDataAdapterType = WellKnownTypes.IDataAdapter(compilationContext.Compilation);
+                commandTextProperty = (IPropertySymbol)iDbCommandType.GetMembers("CommandText").Single();
+            });
+
+            context.RegisterOperationBlockStartAction(operationBlockStartContext =>
```

Comment:
We should register all actions with `compilationContext` instead of `context` as they are accessing state (well known types) computed for a given compilation. Basically make these nested within RegisterCompilationStartAction.

Reply:
Done.